tiledb/common/memory_tracker.cc

```diff
@@ -147,8 +147,12 @@ uint64_t MemoryTrackerResource::get_count() {
 }
 
 void* MemoryTrackerResource::do_allocate(size_t bytes, size_t alignment) {
-  total_counter_.fetch_add(bytes, std::memory_order_relaxed);
+  auto previous_total =
+      total_counter_.fetch_add(bytes, std::memory_order_relaxed);
   type_counter_.fetch_add(bytes, std::memory_order_relaxed);
+  if (previous_total + bytes > memory_budget_ && on_budget_exceeded_) {
```

Comment:
I don't remember if I should call the function every time we make an allocation over the budget, or only every time we make an allocation that crosses the budget.

Reply:
Everytime...

Reply:
Great, that's the existing behavior.